pkg/liquidity-mining/contracts/gauges/ChildChainGauge.vy

```diff
@@ -227,8 +243,14 @@ def _checkpoint_rewards(_user: address, _total_supply: uint256, _claim: bool, _r
                 receiver = _user
 
     reward_count: uint256 = self.reward_count
-    for i in range(MAX_REWARDS):
-        if i == reward_count:
+    reward_indexes: DynArray[uint256, MAX_REWARDS] = []
+    if len(_input_reward_indexes) == 0:
+        reward_indexes = self._all_indexes()
+    else:
+        reward_indexes = _input_reward_indexes
+
+    for i in reward_indexes:
+        if i >= reward_count:
```

Comment:
The input indexes can potentially be higher than the reward count now. In that case the loop will just break, but we could revert as well.

Reply:
Note for the reviewer: I changed my mind when writing tests. There is no reason to allow invalid indexes here, so I changed this check for an assertion which I believe is safer and more clear. Tests were updated accordingly.